src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/InterpolatedStringBuilder.cs

```diff
@@ -54,20 +61,20 @@ private InterpolatedStringBuilder(int initialCapacity)
         private InterpolatedStringBuilder(int initialCapacity, IFormatProvider? provider)
         {
             _provider = provider;
-            _customFormatter = provider?.GetFormat(typeof(ICustomFormatter));
             _chars = _arrayToReturnToPool = ArrayPool<char>.Shared.Rent(initialCapacity);
             _pos = 0;
+            _hasCustomFormatter = provider?.GetFormat(typeof(ICustomFormatter)) != null;
```

Comment:
Wondering if we can avoid the indirection in the common path here. The overwhelmingly common case is that the _provider_ will be a `CultureInfo` instance, and `CultureInfo.GetFormat(typeof(ICustomFormatter))` returns _nullptr_. Might want to investigate the perf of:

```cs
_hasCustomFormatter = provider is not null
    && provider.GetType() != typeof(CultureInfo)
    && provider.GetFormat(typeof(ICustomFormatter)) != null;
```

Reply:
Nice suggestion.  Will do that.